Document n/m/SharedConstants

sciwhiz12 opened the pull request:
This PR documents the `net/minecraft/SharedConstants` class and its fields. Not all fields and methods are mapped, including the `DEBUG_*` fields. I've added details to some javadocs where appropiate, including a note in the class javadoc about the constant inlining and conditional compilation performed by the Java compiler (along with links to the relevant JLS sections, because **:D**) to explain why the constants seem unused.

sciwhiz12 committed:
Document n/m/SharedConstants

sciwhiz12 committed:
Fix other stuff caught by sanitizeData

sciwhiz12 committed:
Remove empty comment

marchermans commented on data/net/minecraft/SharedConstants.mapping:
```diff
@@ -1,12 +1,116 @@
+	COMMENT
+	COMMENT <p><strong>Note:</strong> The majority of the fields within this class are {@code public static final} with constant expressions (constants), and are inlined by the Java compiler at all places which reference these constant fields. Therefore, changing the value of these constant fields will have no effect on already compiled code.</p>
+	COMMENT
+	COMMENT <p>In addition, it is presumed that a large portion of these constant fields (such as those prefixed with {@code DEBUG_} are used as 'flags', for manually toggling code meant for use by Mojang developers in debugging. Therefore, optimizing compilers (which include the Java compiler) may omit the code hidden behind disabled flags, and will result in these flags having to apparent use in t ...
```
having no apparent use.

sciwhiz12 committed:
Fix typo of 'to' to 'no'

Harleyoc1 commented on data/net/minecraft/SharedConstants.mapping:
```diff
@@ -1,12 +1,116 @@
+	COMMENT
+	COMMENT <p>In addition, it is presumed that a large portion of these constant fields (such as those prefixed with {@code DEBUG_} are used as 'flags', for manually toggling code meant for use by Mojang developers in debugging. Therefore, optimizing compilers (which include the Java compiler) may omit the code hidden behind disabled flags, and will result in these flags having no apparent use in t ...
+	COMMENT
+	COMMENT @see <a href="https://docs.oracle.com/javase/specs/jls/se17/html/jls-15.html#jls-15.29">The Java&reg; Language Specification, Java SE 17 Edition, &sect; 15.29. "Constant Expressions"</a>
+	COMMENT @see <a href="https://docs.oracle.com/javase/specs/jls/se17/html/jls-14.html#jls-14.22">The Java&reg; Language Specification, Java SE 17 Edition, &sect; 14.22. "Unreachable Statements"</a>
```
Perhaps link to the J16 spec since that's the version Minecraft is currently compiled in.

sciwhiz12 replied:
Fair point; I've changed them.

sciwhiz12 committed:
Change JLS links to Java SE 16 version

Java 16 is used to compile and run Minecraft 1.17, so the javadocs
should be linking to the specs for Java 16 too.
